lightningd/plugin_hook.c

```diff
@@ -118,10 +118,26 @@ static void plugin_hook_callback(const char *buffer, const jsmntok_t *toks,
 		tal_free(r);
 }
 
+static void plugin_hook_call_next(struct plugin_hook_request *ph_req)
+{
+	struct jsonrpc_request *req;
+	const struct plugin_hook *hook = ph_req->hook;
+	ph_req->current_plugin++;
+	assert(ph_req->current_plugin < tal_count(hook->plugins));
```

Comment:
If a plugin is freed during traversal, this assert will fail.  In general, this problem is hard: even without this, you can skip plugins due to removal at the wrong time.